identity/src/main/java/com/android/identity/storage/StorageEngine.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.identity.storage;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Collection;
+
+/**
+ * A simple interface for storing key/value-pairs.
+ */
+public interface StorageEngine {
+
+    /**
+     * Loads data.
+     *
+     * @param name the name of the file to load.
+     * @return The stored data, as a binary blob or {@code null} if there is no data.
+     */
+    @Nullable byte[] loadData(@NonNull String name);
+
+    /**
+     * Saves data.
+     *
+     * @param name the name of the file to load.
```

Comment:
Yeah, good point. In fact, since the intent is that it's just a key/value store, maybe it would be cleaner just having the methods be `get()`, `put()`, `delete()`, `deleteAll()`, `enumerate()`... let me try and do that and we can see how it works...